### PR TITLE
feat: Add Grafana to syndesis install #3918.

### PR DESCRIPTION
**Pull request by jamesnetherton**
Initial stab at adding Grafana to the install process for #3918. The finer detail for some of the config options will likely need optimizing at some point. But, this PR serves as a baseline for getting something up and running, which others can use when working on related monitoring / metrics tasks.

### CODE REVIEW

File: install/operator/pkg/syndesis/configuration/configuration_test.go [hunk 1]
```diff
@@ -88,6 +88,15 @@ func TestSpecificConfig(t *testing.T) {
 						VolumeCapacity: "7Gi",
 					},
 				},
+				Grafana: v1alpha1.GrafanaConfiguration{
+					Resources: v1alpha1.Resources{
+						ResourceRequirements: v12.ResourceRequirements{
+							Limits: v12.ResourceList{
+								"memory": resource.MustParse("2Gi"),
```
Comment (lgarciaaco) [inline install/operator/pkg/syndesis/configuration/configuration_test.go]: Isn't this around 2Gb?? I would suggest something closer to the 256Mb.

Comment (jamesnetherton) [reply install/operator/pkg/syndesis/configuration/configuration_test.go]: Yeah, it's just a test case so I picked a random value. I can modify it to 256Mb though.

I actually created a new template param `GRAFANA_MEMORY_LIMIT` which defaults to 512Mb which I can also modify to 256Mb if you think it's better.

Comment (lgarciaaco) [reply install/operator/pkg/syndesis/configuration/configuration_test.go]: Oh I see, 512Mb is more reasonable dough ... great stuff

Comment (lgarciaaco) [reply install/operator/pkg/syndesis/configuration/configuration_test.go]: 512Mb makes more sense, great stuff!

**Pull request merged**